virtcontainers/container.go

```diff
@@ -304,6 +304,13 @@ func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (
 			continue
 		}
 
+		// We need to treat /dev/shm as a special case. This is passed as a bind mount in the spec,
+		// but it does not make sense to pass this as a 9p mount from the host side.
+		// This needs to be handled purely in the guest, by allocating memory for this inside the VM.
+		if m.Destination == "/dev/shm" {
```

Comment:
Both `runc spec` and `docker-runc spec` creates an OCI spec listing `/dev/shm` as `tmpfs`. How did you get the bind mounted `/dev/shm` in OCI spec?

Reply:
@bergwolf 
It looks like docker will create a bind mount for "/dev/shm", I just checked config file from docker, it adds this item:
```
{"destination":"/dev/shm","type":"bind","source":"/var/lib/docker/containers/fc243bca3002a1264072d07086b77970ab9e6a8da9c5143b0af6c0fbb1f2367d/shm","options":["rbind","rprivate"]}
```
So this change makes sense in this case.

Reply:
Default mounts are here:

```
    "mounts": [
        {
            "destination": "/proc",
            "options": [
                "nosuid",
                "noexec",
                "nodev"
            ],
            "source": "proc",
            "type": "proc"
        },
        {
            "destination": "/dev",
            "options": [
                "nosuid",
                "strictatime",
                "mode=755"
            ],
            "source": "tmpfs",
            "type": "tmpfs"
        },
        {
            "destination": "/dev/pts",
            "options": [
                "nosuid",
                "noexec",
                "newinstance",
                "ptmxmode=0666",
                "mode=0620",
                "gid=5"
            ],
            "source": "devpts",
            "type": "devpts"
        },
        {
            "destination": "/sys",
            "options": [
                "nosuid",
                "noexec",
                "nodev",
                "ro"
            ],
            "source": "sysfs",
            "type": "sysfs"
        },
        {
            "destination": "/sys/fs/cgroup",
            "options": [
                "ro",
                "nosuid",
                "noexec",
                "nodev"
            ],
            "source": "cgroup",
            "type": "cgroup"
        },
        {
            "destination": "/dev/mqueue",
            "options": [
                "nosuid",
                "noexec",
                "nodev"
            ],
            "source": "mqueue",
            "type": "mqueue"
        },
        {
            "destination": "/etc/resolv.conf",
            "options": [
                "rbind",
                "rprivate"
            ],
            "source": "/var/lib/docker/containers/fc243bca3002a1264072d07086b77970ab9e6a8da9c5143b0af6c0fbb1f2367d/resolv.conf",
            "type": "bind"
        },
        {
            "destination": "/etc/hostname",
            "options": [
                "rbind",
                "rprivate"
            ],
            "source": "/var/lib/docker/containers/fc243bca3002a1264072d07086b77970ab9e6a8da9c5143b0af6c0fbb1f2367d/hostname",
            "type": "bind"
        },
        {
            "destination": "/etc/hosts",
            "options": [
                "rbind",
                "rprivate"
            ],
            "source": "/var/lib/docker/containers/fc243bca3002a1264072d07086b77970ab9e6a8da9c5143b0af6c0fbb1f2367d/hosts",
            "type": "bind"
        },
        {
            "destination": "/dev/shm",
            "options": [
                "rbind",
                "rprivate"
            ],
            "source": "/var/lib/docker/containers/fc243bca3002a1264072d07086b77970ab9e6a8da9c5143b0af6c0fbb1f2367d/shm",
            "type": "bind"
        }
    ],
```

In this bind mount case, ignoring `shm` could be enough.

Reply:
Thanks @WeiZhang555 !